ext/standard/info.c

```diff
@@ -660,6 +660,8 @@ static void php_get_windows_cpu(char *buf, size_t bufsize)
 PHPAPI zend_string *php_get_uname(char mode)
 {
 	char *php_uname;
+
+	ZEND_ASSERT(mode == 'a' || mode == 'm' || mode == 'n' || mode == 'r' || mode == 's' || mode == 'v');
```

Comment:
*Might* use `strchr()` here (and below), and define the string of allowed characters as macro. This way any possible extensions in the future might not break this assertion.

Reply:
I added a static inline function to check that the value is a valid mode

Reply:
That's even better!